starknet_py/utils/typed_data/types.py

```diff
@@ -0,0 +1,18 @@
+from dataclasses import dataclass
+from typing import Optional, Union
+
+from starknet_py.utils.typing import TypedDict
+
+
+class StarkNetDomain(TypedDict):
+    name: Optional[str]
+    version: Optional[str]
+    chainId: Optional[Union[str, int]]
+
+
+@dataclass(frozen=True)
+class TypedData:
+    types: dict
```

Comment:
Types have know typing `Dict[str, List[Dict[str, str]]]`so this could have better typing. 

I'd actually consider defining more data classes and making this typed more clearly. We could then use Marshmallow schema to accept typeddata as dict or typeddict from users (so it's convenient for them) and then parsing it to classes that are easy for us to operate on.